File: flutter-internal/lib/models/session_context.dart
import 'dart:typed_data';

/// In-memory ECDH-derived session state.
class SessionContext {
  final String sessionId;
  final Uint8List sessionKey; // 32-byte AES-256 key
  final String kid;
  final String clientId;
  final bool authenticated;
  final int expiresInSec;

  SessionContext({
    required this.sessionId,
    required this.sessionKey,
    required this.kid,
    required this.clientId,
    required this.authenticated,
    required this.expiresInSec,
  });

  /// Securely zero the session key in memory.
  void zeroize() {
    for (var i = 0; i < sessionKey.length; i++) {
      sessionKey[i] = 0;
    }
  }
}

File: dotnet-internal/Models/SessionContext.cs
using System.Security.Cryptography;

namespace IdentityExample.Models;

/// <summary>
/// Holds the ECDH-derived session state after key exchange.
/// </summary>
public sealed class SessionContext
{
    public string SessionId { get; }
    public byte[] SessionKey { get; private set; }
    public string Kid { get; }
    public string ClientId { get; }
    public bool Authenticated { get; }
    public int ExpiresInSec { get; }

    public SessionContext(string sessionId, byte[] sessionKey, string kid,
                          string clientId, bool authenticated, int expiresInSec)
    {
        SessionId = sessionId;
        SessionKey = sessionKey;
        Kid = kid;
        ClientId = clientId;
        Authenticated = authenticated;
        ExpiresInSec = expiresInSec;
    }

    public void Zeroize()
    {
        if (SessionKey != null)
        {
            CryptographicOperations.ZeroMemory(SessionKey);
            SessionKey = null!;
        }
    }
}
